src/instructlab/model/list.py

```diff
@@ -34,32 +38,35 @@ def model_list(model_dirs: list[str], list_checkpoints: bool):
     """lists models"""
     data = []
     for directory in model_dirs:
-        for entry in Path(directory).iterdir():
-            # if file, just tally the size. This must be a GGUF.
-            if entry.is_file() and is_model_gguf(entry):
-                model, age, size = _analyze_gguf(entry)
-                data.append([model, age, size])
+        d = Path(directory)
+        if not d.is_dir():
+            continue
+        for entry in d.iterdir():
+            if is_model_gguf(entry):
+                data.append(_analyze_gguf(entry))
             elif entry.is_dir():
-                for m in _analyze_dir(entry, list_checkpoints, directory):
-                    data.append(m)
-    print_table(["Model Name", "Last Modified", "Size"], data)
+                data += list(_analyze_dir(entry, list_checkpoints, directory))
+    print_table(("Model Name", "Last Modified", "Size"), data)
 
 
-# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MiB, GiB)
-def convert_bytes_to_proper_mag(f_size: float):
-    magnitudes = ["KiB", "MiB", "GiB", "TiB"]
+def convert_bytes_to_human_readable_size(n_bytes: int) -> str:
```

Comment:
We shouldn't do this because it couples the implementation of this function too tightly to where it gets used. This will make things more complicated down the road. We should instead just return a float along with the magnitude representation, and then let the caller do whatever with it as they like. 
The other problem here is that it assumes what the display decides to round the number to. What if we want to do 2 floating points in one place, but actually check the whole value somewhere else? Now we need to add in an additional `precision` arg, etc.

Reply:
What I'm trying to avoid here is that this code in this module duplicates this conversion and calculation. The part that produces `Tuple[float, str]` may be spun out into a separate function for others' use (I assume you'd like to see it in `utils.py`?)